File: 100-199/169-majority-element.py
# coding: utf-8

class Solution:
    # @param num, a list of integers
    # @return an integer
    def majorityElement(self, num):
        dic = {}
        for n in num:
        	if dic.has_key(n):
        		dic[n] += 1
        	else:
        		dic[n] = 1
        l = len(num)
        for n, count in dic.items():
        	if count > l/2:
        		return n

File: 100-199/169-majority-element.go
package main

func majorityElement(nums []int) int {
	var (
		mode    = nums[0]
		counter = 1
	)
	for _, num := range nums[1:] {
		if num == mode {
			counter++
		} else {
			counter--
		}
		if counter < 0 {
			mode = num
			counter = 1
		}
	}
	return mode
}
